backend/src/meetings/meetings.service.ts

```diff
@@ -218,11 +221,127 @@ export class MeetingsService {
     return this.bookingRepository.save(await this.bookingRepository.find({ where: { mid: id }, relations: {user: true}}));
   }
 
+  async confirmUserForMeeting(id: number, username: string): Promise<BookingEntity>{
+    username = username.toLowerCase();
+    const booking = await this.bookingRepository.findOne({
+      where: {mid: id, user: {username: username}, deleted: false},
+      relations: {user: true}
+    });
+    if (!booking) {
+      throw new HttpException('Booking not found', HttpStatus.NOT_FOUND);
+    }
+    booking.tookPart = true;
+    booking.dateOfConfirmation = new Date();
+    return this.bookingRepository.save(booking);
+  }
+
+  async rejectUserFromMeeting(id: number, username: string): Promise<BookingEntity>{
+    username = username.toLowerCase();
+    const booking = await this.bookingRepository.findOne({
+      where: {mid: id, user: {username: username}, deleted: false},
+      relations: {user: true}
+    });
+    if(!booking){
+      throw new HttpException('Booking not found', HttpStatus.NOT_FOUND);
+    }
+    booking.tookPart = false;
+    return this.bookingRepository.save(booking);
+  }
+
+  async attendingToMeeting(mid: number, username: string){
+    username = username.toLowerCase();
+    const booking = await this.bookingRepository.findOne({
+      where: {mid: mid, user: {username: username}},
+      relations: {user: true}
+    });
+    if (!booking){
+      const newBooking = new BookingEntity();
+      newBooking.mid = mid;
+      const user = await this.usersService.findOne(username);
+      newBooking.user = user;
+      newBooking.tookPart = false;
+      newBooking.dateOfRegistration = new Date();
+      newBooking.externals = [''];
+      newBooking.deleted = false;
+      const savedBooking = await this.bookingRepository.save(newBooking);
+      const meeting = await this.meetingRepository.findOne({where: {mid: mid}});
+      if (meeting && savedBooking.user) {
+        this.notificationAboutMeetingsService.notifyAuthor(true, meeting, savedBooking.user);
+      }
+      return savedBooking;
+    }
+    return booking;
+  }
+
+  async notAttendingToMeeting(mid: number, username: string): Promise<BookingEntity>{
+    username = username.toLowerCase();
+    const booking = await this.bookingRepository.findOne({
+      where: {mid: mid, user: {username: username}, deleted: false},
+      relations: {user: true}
+    });
+    if (!booking){
+      throw new HttpException('Booking not found', HttpStatus.NOT_FOUND);
+    }
+    booking.deleted = true;
+    const meeting = await this.meetingRepository.findOne({where: {mid: mid}});
+    const savedBooking = await this.bookingRepository.save(booking);
+    if(meeting && booking.user){
+      if (meeting.username == booking.user.username){
```

Comment:
The code is not correct. Compare again with the server. The idea is that when the "booking" is removed by the author, the participant who was removed should be notified. If it is the participant himself, the author should be notified.